services/src/unit-test/java/uk/gov/moj/sdt/services/TargetApplicationSubmissionServiceTest.java

```diff
@@ -145,160 +154,238 @@ public void setUp() {
+        individualReqProcessingDelay.setName(MCOL_INDV_REQ_DELAY);
+        when(this.mockCacheable.getValue(IGlobalParameter.class, MCOL_INDV_REQ_DELAY)).thenReturn(
+            individualReqProcessingDelay);
+
+        individualRequest.setRequestStatus(FORWARDED);
+
+        final IGlobalParameter connectionTimeOutParam = new GlobalParameter();
+        connectionTimeOutParam.setName(TARGET_APP_TIMEOUT);
+        connectionTimeOutParam.setValue("1000");
+        when(this.mockCacheable.getValue(IGlobalParameter.class, TARGET_APP_TIMEOUT)).thenReturn(
+            connectionTimeOutParam);
+
+        final IGlobalParameter receiveTimeOutParam = new GlobalParameter();
+        receiveTimeOutParam.setName(TARGET_APP_RESP_TIMEOUT);
+        receiveTimeOutParam.setValue(TWELVE_THOUSAND);
+        when(this.mockCacheable.getValue(IGlobalParameter.class, TARGET_APP_RESP_TIMEOUT)).thenReturn(
+            receiveTimeOutParam);
+
+        doAnswer(invocation -> {
+            IndividualRequest argument = invocation.getArgument(0);
+            argument.setRequestStatus(IIndividualRequest.IndividualRequestStatus.ACCEPTED.getStatus());
+            return null;
+        }).when(mockConsumerGateway).individualRequest(individualRequest, 1000, 12000);
+
+        final IBulkSubmission bulkSubmission = individualRequest.getBulkSubmission();
+
+        when(this.mockIndividualRequestDao.queryAsCount(eq(IndividualRequest.class), isA(
+            Supplier.class))).thenReturn(0L);
 
         // Setup dummy target response
         SdtContext.getContext().setRawInXml("response");
 
         this.targetAppSubmissionService.processRequestToSubmit(sdtRequestRef);
 
-        EasyMock.verify(mockIndividualRequestDao);
-        EasyMock.verify(mockConsumerGateway);
-        EasyMock.verify(mockCacheable);
+        List<ILoggingEvent> logList = listAppender.list;
 
-        Assert.assertEquals("Bulk submission status is incorrect", IBulkSubmission.BulkRequestStatus.COMPLETED
-                .getStatus(), individualRequest.getBulkSubmission().getSubmissionStatus());
-        Assert.assertNotNull("Bulk submission completed date should be populated", individualRequest
-                .getBulkSubmission().getCompletedDate());
-        Assert.assertNotNull("Bulk submission updated date should be populated", individualRequest
-                .getBulkSubmission().getUpdatedDate());
+        assertTrue(logList.isEmpty(),"Logging set to only Error");
 
+        logger.detachAndStopAllAppenders();
+
+        verify(mockIndividualRequestDao, times(2)).persist(individualRequest);
     }
 
+
+
     /**
      * This method checks an all positive scenario for processing request to submit.
      * In the end there are still individual requests outstanding so the bulk submission
      * is not marked as completed.
      */
     @Test
     public void processRequestToSubmitSuccess() {
-        final String sdtRequestRef = "TEST_1";
+        final String sdtRequestRef = TEST_1;
```

Comment:
The sdtRequestRef variable is redundant and can be removed.  The TEST_1 constant can be used in it's place.

Reply:
done